torchaudio/transforms.py

```diff
@@ -9,6 +9,9 @@
 from torchaudio import functional as F
 from torchaudio.compliance import kaldi
 
+import numpy as np
+from scipy import special
```

Comment:
Hi @small-yellow-duck 

Thanks for opening a PR. I will try this out to think how to test. I have not looked into the detail yet but `numpy` and `scipy` are not a mandatory dependencies of `torchaudio`.

We have a mechanism to differ loading the optional dependencies but it would be nice if we can implement without them. Quick googling tells that `torch` has `heaviside` function, so removing `numpy` appears plausible. We need to think what to do for `scipy` dependency.

Reply:
`scipy.special.i0` is used to compute the kaiser window for `kaiser`, `kaiser_best` and `kaiser_fast`. If scipy dependency is not desirable, then one option would be to just allow `hann` window mode. (the original goal of including `kaiser` windows was to try to duplicate the behaviour of librosa, but I don't think the results of librosa and this proposed routine are identical to numeric precision, so there is likely some other detail about how the librosa filters were generated that I haven't managed to duplicate). If the routine only needs to support `hann` windows, then I agree that it should be possible to remove numpy and scipy as dependencies.

I also see that there is an issue suggesting that pytorch add the modified bessel functions, although there doesn't seem to be a PR to merge the code into the pytorch codebase: https://github.com/pytorch/pytorch/issues/7815 . If the code to calculate the equivalent of `scipy.special.i0` is added to pytorch, then it should also be possible to include `kaiser` windows.

Reply:
I realized that sinc can be written in terms of the gamma function - and pytorch has a routine to compute the log of the gamma function (`torch.lgamma`)! So I think it should be possible to excise numpy as a dependency as well.... (see output 10 in https://mathworld.wolfram.com/SincFunction.html)